projects/meta_cl/experiments/dendrites.py

```diff
@@ -0,0 +1,298 @@
+        """
+        prepended = {}
+        for n, p in self.prediction.named_parameters():
+            n = "prediction." + n
+            prepended[n] = p
+        for n, p in self.gating_layer.named_parameters():
+            n = "gating_layer." + n
+            prepended[n] = p
+        return prepended
+
+    @property
+    def named_slow_params(self):
+        """
+        Returns named params of adaption network, being sure to prepend
+        the names with "representation."
+        """
+        prepended = {}
+        for n, p in self.modulation.named_parameters():
+            n = "modulation." + n
+            prepended[n] = p
+        return prepended
+
+    def forward(self, x):
+        mod = self.modulation(x)
+        pred = self.prediction(x)
+        out = self.gating_layer(pred, context=mod)
+        return out
+
+
+# Alternative to run on a single GPU
+def run_experiment(config):
+    exp = config.get("experiment_class")()
+    exp.setup_experiment(config)
+    print(f"Training started....")
+    while not exp.should_stop():
+        result = exp.run_epoch()
+        print(f"Accuracy: {result['mean_accuracy']:.4f}")
+    print(f"....Training finished")
+
+
+# Simple Omniglot Experiment
+metacl_base = dict(
+    # training infrastructure
+    ray_trainable=SupervisedTrainable,
+    distributed=False,
+    # problem specific
+    experiment_class=MetaContinualLearningExperiment,
+    dataset_class=omniglot,
+    model_class=OMLNetwork,
+    model_args=dict(input_size=(105, 105)),
+    # metacl variables
+    num_classes=963,
+    num_classes_eval=660,
+    batch_size=5,
+    val_batch_size=15,
+    slow_batch_size=64,
+    replay_batch_size=64,
+    epochs=1000,
+    tasks_per_epoch=10,
+    adaptation_lr=0.03,
+    # generic
+    optimizer_class=torch.optim.Adam,
+    optimizer_args=dict(lr=1e-4),
+    dataset_args=dict(root="~/nta/datasets"),
+    local_dir=os.path.expanduser("~/nta/results/experiments/meta_cl"),
+)
+
+metacl_oml = copy.deepcopy(metacl_base)
+metacl_oml.update(
+    experiment_class=OMLExperiment,
+    run_meta_test=True,
+)
+
+# This attempts to replicate the results from the original OML paper.
+# See results below.
+metacl_oml_replicate = copy.deepcopy(metacl_oml)
+metacl_oml_replicate.update(
+
+    # How often to checkpoint (in epochs)
+    checkpoint_freq=1000,
+    keep_checkpoints_num=1,
+    checkpoint_at_end=True,
+    checkpoint_score_attr="training_iteration",
+
+    # Number of tasks (i.e. distinct classes) for the inner loops and outer loop
+    tasks_per_epoch=3,
+
+    # Batch size for each inner and outer step, respectively.
+    batch_size=1,       # use 1 images per step w/in the inner loop
+    slow_batch_size=5,  # use 5 images per class w/in the outer loop.
+
+    # Number of steps per task within the inner loop.
+    num_fast_steps=5,  # => we take 5 * 'tasks_per_epoch' sequential grad steps
+    # num_slow_steps=1,  # not actually defined, but technically this is always one.
```

Comment:
If we don't need `num_slow_steps`, then let's remove it.

Reply:
I would generally agree if it was not for an experiments file. We are usually ok with commented code in experiments file. It serves as documentation, and it is common that you are switching things on and off during experiments.